lib-impl/src/main/java/org/ergoplatform/appkit/impl/OutBoxBuilderImpl.scala

```diff
@@ -39,11 +39,14 @@ class OutBoxBuilderImpl(_txB: UnsignedTransactionBuilderImpl) extends OutBoxBuil
     val tokenNumOfDecVal = token.getMintingBoxR6
     _registers ++= Array(tokenNameVal, tokenDescVal, tokenNumOfDecVal)
 
-    // optional registers, but either all of them or none
-    if (token.getMintingBoxR7 != null && token.getMintingBoxR8 != null) {
-      _registers ++= Array(token.getMintingBoxR7, token.getMintingBoxR8)
-      if (token.getMintingBoxR9 != null)
-        _registers += token.getMintingBoxR9
+    if(token.getMintingBoxR7 != null){
+      _registers ++= Array(token.getMintingBoxR7)
+    }
+    if(token.getMintingBoxR8 != null){
+      _registers ++= Array(token.getMintingBoxR8)
+    }
+    if(token.getMintingBoxR9 != null){
+      _registers ++= Array(token.getMintingBoxR9)
```

Comment:
There is a potential problem here. When R8 == null, then the value of token.getMintingBoxR9 will be added after R7 and will lend into R8. There will be a lot of confusion and hard to debug.
So, R9 cannot be added is R8 is null. 
The same for R7 (when it is null, R8 and R9 cannot be added).
This constrants better be checked early on.

Reply:
Good point, please see the fix.